src/backend/services/file.py

```diff
@@ -24,11 +31,267 @@
+        )
+        if not conversation:
+            raise HTTPException(
+                status_code=404,
+                detail=f"Conversation with ID: {conversation_id} not found.",
+            )
+        file_ids = conversation.file_ids
+
+        files = []
+        if file_ids is not None:
+            files = file_crud.get_files_by_ids(session, file_ids, user_id)
+
+        return files
+
+    def delete_file_from_conversation(
+        self, session: DBSessionDep, conversation_id: str, file_id: str, user_id: str
+    ) -> None:
+        """
+        Delete file from conversation
+
+        Args:
+            session (DBSessionDep): The database session
+            conversation_id (str): The conversation ID
+            file_id (str): The file ID
+            user_id (str): The user ID
+        """
+        conversation_crud.delete_conversation_file_association(
+            session, conversation_id, file_id, user_id
+        )
+        file_crud.delete_file(session, file_id, user_id)
+        return
+
+    def get_file_by_id(self, session: DBSessionDep, file_id: str, user_id: str) -> File:
+        """
+        Get file by ID
+
+        Args:
+            session (DBSessionDep): The database session
+            file_id (str): The file ID
+            user_id (str): The user ID
+
+        Returns:
+            File: The file that was created
+        """
+        file = file_crud.get_file(session, file_id, user_id)
+        return file
+
+    def get_files_by_ids(
+        self, session: DBSessionDep, file_ids: list[str], user_id: str
+    ) -> list[FileModel]:
+        """
+        Get files by IDs
+
+        Args:
+            session (DBSessionDep): The database session
+            file_ids (list[str]): The file IDs
+            user_id (str): The user ID
+
+        Returns:
+            list[File]: The files that were created
+        """
+        files = file_crud.get_files_by_ids(session, file_ids, user_id)
+        return files
+
+    def update_file(
+        self, session: DBSessionDep, file: File, new_file: UpdateFileRequest
+    ) -> File:
+        """
+        Update file
+
+        Args:
+            session (DBSessionDep): The database session
+            file (File): The file to update
+            new_file (UpdateFileRequest): The new file data
+
+        Returns:
+            File: The updated file
+        """
+        updated_file = file_crud.update_file(session, file, new_file)
+        return updated_file
+
+    def bulk_delete_files(
+        self, session: DBSessionDep, file_ids: list[str], user_id: str
+    ) -> None:
+        """
+        Bulk delete files
+
+        Args:
+            session (DBSessionDep): The database session
+            file_ids (list[str]): The file IDs
+            user_id (str): The user ID
+        """
+        file_crud.bulk_delete_files(session, file_ids, user_id)
+
+    def get_message_files(
+        self, session: DBSessionDep, message_id: str, user_id: str
```

Comment:
I would rename this to `get_files_by_message` or message_id. 

I suggest implementing a batch get as well that takes as input a List of message_ids, and using that instead where needed

Reply:
I'm assuming this is also related to https://github.com/cohere-ai/cohere-toolkit/pull/451#discussion_r1692215784, so I'll address these related requests here.

We want to use the method to get file(s) _per_ message. So I can't see how using a batch get files with a list of message_ids would really work here.

Reply:
Got it! Sgtm